### DISCUSSION TITLE
Is it correct that the App gets installed and uninstalled every time?

### DISCUSSION THREAD

**Discussion by Spacewalker2** [mrrfv/open-android-backup#50] [Q&A]
Hello,

it seems LineageOS 20 is more restrictive and I have to grant
permissions for call log, SMS and the storage separately. This can only
be accomplished during the time windows when the app is available and
has to be done in the Android settings every time again.

Thanks for this great project!

**Comment by mrrfv** [comment]
The companion app is only installed for the time when it's needed, so it's true that the app is getting immediately uninstalled after finishing what it's supposed to do.

The script should automatically grant all required permissions (aside from the "full internal storage access" one, required on newer Android versions) to the app, so you shouldn't have to do that manually. Are you seeing any error messages when the permissions are being granted?

**Reply by Spacewalker2** [reply]
I observed it in a way that when I pushed the "Export data" button and didn't grant anything I get an error inside the app. When the app appears I have to go inside the Android app settings for the companion app and allow the access for the call log and SMS. The "full internal storage access" appears when the app is launched, but the app isn't started by it the script every time. For at least once I had to launch it by myself. If I don't do these steps the script is printing errors of not having those permissions and tries to go on. I'm happy to test this further. Maybe there is some additional advice. Would not uninstalling the app and preserving the former granted permissions maybe solve this issue?

**Reply by mrrfv** [reply]
Could you try a different cable or using the wireless connection mode? It might just be that the device is being disconnected while the script's trying to grant permissions to the companion app.

> Would not uninstalling the app and preserving the former granted permissions maybe solve this issue?

The app is automatically removed so that it won't be included in the backup archive (the app is first removed, and then all apps are getting pulled off the device) and because it serves no use after doing what it's supposed to do.
